libraries/chain/include/eosio/chain/snapshot.hpp

```diff
@@ -5,6 +5,7 @@
 #include <fc/variant_object.hpp>
 #include <boost/core/demangle.hpp>
 #include <ostream>
+#include <memory>
 
 namespace eosio { namespace chain {
```

Comment:
Please rebase this PR to main (or start a new PR) and incorporate `libraries/chain/include/eosio/chain/snapshot.hpp` & `libraries/chain/snapshot.cpp` changes from https://github.com/AntelopeIO/leap/pull/24 into it. Then we will be able to get the PR into `main`. The `--snapshot-to-json` portion of #24 can be added to `leap-util` later (keep that PR up so it can be easily incorporated into `leap-util`.